### PR TITLE
Utilise the migrate template API when applying changes to the deployment template id 

### PR DESCRIPTION
**Pull request by tobio**
## Description

Use the migrate template API to build the base update request when an existing deployment changes.  `deployment_template_id`, 

## Related Issues
Fixes https://github.com/elastic/terraform-provider-ec/issues/487
Fixes https://github.com/elastic/terraform-provider-ec/issues/556

## Motivation and Context

This API is purpose built to update deployment options to be compatible with a different template and simplifies the process for migrating deployments between hardware types. 

## How Has This Been Tested?

Manually, acceptance test, unit tests

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: ec/ecresource/deploymentresource/expanders_test.go [hunk 1]
```diff
@@ -3767,15 +3716,14 @@ func Test_updateResourceToModel(t *testing.T) {
 				},
 			},
 		},
-		// The behavior of this change should be:
-		// * Resets the Elasticsearch topology: from 16g (due to unsetTopology call on DT change).
-		// * Keeps the kibana toplogy size to 2g even though the topology element has been removed (saved value persists).
-		// * Removes all other non present resources
+		// // The behavior of this change should be:
+		// // * Keeps the kibana toplogy size to 2g even though the topology element has been removed (saved value persists).
+		// // * Removes all other non present resources
```
Comment (webfella) [inline ec/ecresource/deploymentresource/expanders_test.go:3721]: yo dawg

**Review by webfella — APPROVED** [review]
All LGTM mate, can't help out with any code suggestions but I def learned a bit while workin through this 👍

File: ec/ecresource/deploymentresource/expanders_test.go (hunk 1 above)
Comment (tobio) [reply ec/ecresource/deploymentresource/expanders_test.go:3721]: Gotta be extra sure

File: ec/acc/testdata/deployment_basic_defaults_hw_2.tf [hunk 2]
```diff
@@ -9,7 +9,16 @@ resource "ec_deployment" "defaults" {
   version                = data.ec_stack.latest.version
   deployment_template_id = "%s"
 
-  elasticsearch {}
+  elasticsearch {
```
Comment (dimuon) [inline ec/acc/testdata/deployment_basic_defaults_hw_2.tf:12]: Does it make sense to modify test checks instead? It looks like the acceptance test (`TestAccDeployment_basic_defaults_hw`) checks updating configuration once the template is changed.
 
Also shall the acceptance test work as expected without these explicit changes to the sizes? The new template (`aws-hot-warm-v2`) has size 4g for both `hot_content` and `warm` tiers so I guess once the template attribute is set to `aws-hot-warm-v2`, the tier sizes should change accordingly even without these explicit settings in the configuration.

**Pull request merged**